### ISSUE TITLE
Used items on Fence cost full price

### ISSUE THREAD

**Issue by refringe** [sp-tarkov/server#447]
## **General information**
**Project issue is for:**
Server

**Expected result**:
Used items should have their price adjusted based on how many uses are left on fence assort, like on live

**Actual result**:
Used items on fence have full price, even though there are items with varied uses left.
This seems to apply to meds and keys, maybe armour to an extent

Some Additional notes:
I have tested on regular BE test.
I have also tested on a copy with a modified fence stash size as to showcase the issue more clearly. It also seems to affect keys but I haven't been able to find any on fence to screenshot

## **How to reproduce**
1. Go to fence
2. Observe items with uses (meds or keys) and their price
3. Can also compare price by filtering item on flea and seeing how full price it is
4. Full price for used keys/meds/armours(?)

## Screenshots
Regular fence
![image](https://dev.sp-tarkov.com/attachments/6e08fb1c-d360-469d-aeb5-3a96a083c7b1)

Adjusted fence
![image](https://dev.sp-tarkov.com/attachments/0482377c-99c4-4006-98c2-95f7997288bc)

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/6e08fb1c-d360-469d-aeb5-3a96a083c7b1) (660 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/0482377c-99c4-4006-98c2-95f7997288bc) (786 KiB)
- [server-2024-02-18.log](https://dev.sp-tarkov.com/attachments/80c1bd4f-483d-4b30-9bc4-4d37b82fb99c) (801 KiB)
- [LogOutput.log](https://dev.sp-tarkov.com/attachments/cd7bae70-a87c-413c-8d15-6dba29076218) (9 KiB)
- [2024.02.18_12-18-25_0.14.1.0.28744 traces.log](https://dev.sp-tarkov.com/attachments/fc853537-9263-461e-bd46-f09196839a5c) (40 KiB)

---

[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508) by _**CrocodileJonesy**_

**Comment by refringe** [comment]
Do you have any live fence data we can compare again?

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23761) by _**chomp**_

**Comment by refringe** [comment]
![image](https://dev.sp-tarkov.com/attachments/ba85c4be-1cfc-44a8-977b-f04ea0f93a33)

I spent 10 minutes refreshing fence on live and can't many of the items to compare against

Painkillers 1/4 = 1853, 4/4 = 7410
ai-2 97/100 = 6558, 100/100 = 6760
army bandage 1/2 = 1332, 2/2 = 2664
salewa 219/400 = 13426, 391/400 = 23971

ifak 300/300 = 30894
golden star 10/10 = 37556
etg-c stim = 49179

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/ba85c4be-1cfc-44a8-977b-f04ea0f93a33) (241 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23762) by _**chomp**_

**Comment by refringe** [comment]
Just got an ai-2
![image](https://dev.sp-tarkov.com/attachments/6de2dfad-d50e-4039-876d-f223f7125cb9)

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/6de2dfad-d50e-4039-876d-f223f7125cb9) (58 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23763) by _**chomp**_

**Comment by refringe** [comment]
![image](https://dev.sp-tarkov.com/attachments/77d042ab-02d9-4961-9dae-8b09f40318c1)
Here is a comparison from bandit that showed this for bandages
I will try looking for more items to compare against

![image](https://dev.sp-tarkov.com/attachments/78e9f805-845f-4f2e-be04-7c4d4730f806)
Here's another screen of fence from live (see version in bottom left)

![image](https://dev.sp-tarkov.com/attachments/c1da6d53-1220-419c-8b95-053780c936a9)
It shows ai-2, painkillers (1/4 uses) and army bandages (1/2 uses) at about 1/2 the price of the regular ones that are full priced

I've defs otherwise seen other items be cheaper but don't have and screen shots of it sadly, will try looking for other meds/keys

EDIT:
![image](https://dev.sp-tarkov.com/attachments/d07d636f-7549-4dfd-a4cd-2d49cf8e2c7a)
Got one for a almost full salewa on live difference is abou 1k RUB but in comparison to spt fence all of the salewas are full price even if very low uses left

EDIT2:
![image](https://dev.sp-tarkov.com/attachments/575b3eac-1f50-4294-8256-d60e7c6ef991)
![image](https://dev.sp-tarkov.com/attachments/d6136556-cf57-430b-af3d-cfe8f99bc50e)
Got another for about half used salewas, price much more visible based on the uses, can't find any keys or other items with uses but hopefully it is enough to go on.
I also included one for armour repair kit that is empty, since I suspect it may affect on spt fence as well

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/77d042ab-02d9-4961-9dae-8b09f40318c1) (31 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/56ac5f24-cd09-4d13-8ea1-b8d2fdc8168d) (881 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/78e9f805-845f-4f2e-be04-7c4d4730f806) (881 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/c1da6d53-1220-419c-8b95-053780c936a9) (1537 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/d07d636f-7549-4dfd-a4cd-2d49cf8e2c7a) (614 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/575b3eac-1f50-4294-8256-d60e7c6ef991) (1343 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/d6136556-cf57-430b-af3d-cfe8f99bc50e) (902 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23764) by _**CrocodileJonesy**_

**Comment by refringe** [comment]
Made changes for medical items: https://dev.sp-tarkov.com/SPT-AKI/Server/commit/7406ee389e2bb0cd55980e9338a4b7bc8c812b29

![image](https://dev.sp-tarkov.com/attachments/f4a244bd-c5ba-4d40-94eb-b4f7b0f00462)

---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/f4a244bd-c5ba-4d40-94eb-b4f7b0f00462) (653 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23769) by _**chomp**_

**Comment by refringe** [comment]
Still to test:

weapons (probably okay?)
Armors (no clue if they're even remotely accurate vs live)

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23772) by _**chomp**_

**Comment by refringe** [comment]
Did some more testing of fence items, I have included screenshots below for reference

Confirmed working for meds on SPT fence
![image](https://dev.sp-tarkov.com/attachments/367aba16-2e97-4c81-91e0-694485c19041)

Keys are still same price on SPT fence (I defs remember seeing price adjusting with uses on live, might need some more confirmation though)
![image](https://dev.sp-tarkov.com/attachments/84e6cf4b-cc1c-4cb7-b2e6-cb17c1b39395)

Some armour plates still the same price on SPT fence
![image](https://dev.sp-tarkov.com/attachments/84d189ac-8e29-429b-ae73-2391386724ad)

Armours from live fence, there is a small difference
![image](https://dev.sp-tarkov.com/attachments/6f89b2a2-0f11-444b-b2bd-41ae4142484a)
![image](https://dev.sp-tarkov.com/attachments/97344263-222b-46e6-b810-a2aeeca5705b)

Armour from SPT, there are durability differences and price is similar, I think this is ok (Might need more confirmation)
![image](https://dev.sp-tarkov.com/attachments/a6c1cbb9-2523-439c-b09d-7dde702060df)

Helmets and some armour plates from SPT fence seems ok, price difference is small, like on live. My guess is price is determined from just max durability of armour plates hence why prices might be different instead of current durability / max durability
![image](https://dev.sp-tarkov.com/attachments/959e57dc-67a4-40ae-80de-71654f4c8954)

I had a look on weapons price difference is hard to tell due to how randomised each weapon is with parts durability etc. I would say it seems ok


**TL;DR**
My conclusion: weapons, armour vests/rigs and helmets seem to be ok so I don't think there is a need to apply price scaling since it already seems to do that. 
Keys, Armour plates (on their own) and possibly provisions (with uses) seem to still ignore use count for price calculation


---
### Attachments
- [image.png](https://dev.sp-tarkov.com/attachments/367aba16-2e97-4c81-91e0-694485c19041) (763 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/84d189ac-8e29-429b-ae73-2391386724ad) (1308 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/6f89b2a2-0f11-444b-b2bd-41ae4142484a) (1279 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/959e57dc-67a4-40ae-80de-71654f4c8954) (1334 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/97344263-222b-46e6-b810-a2aeeca5705b) (994 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/a6c1cbb9-2523-439c-b09d-7dde702060df) (1576 KiB)
- [image.png](https://dev.sp-tarkov.com/attachments/84e6cf4b-cc1c-4cb7-b2e6-cb17c1b39395) (1425 KiB)


---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23862) by _**CrocodileJonesy**_

**Comment by refringe** [comment]
When we was investigating armor plates being too cheap we noticed BSG was using the max durability value to determine the price change. Armor plates now use item max durability / itemDB max durability to get the discount %.

We are currently making a spreadsheet of Live flea items.

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23863) by _**Kaeno**_

**Comment by refringe** [comment]
Ah ok, I wasn't sure about armours or helmets and such, only certain about meds and keys with price adjustments

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23864) by _**CrocodileJonesy**_

**Comment by refringe** [comment]
Just as we are now gathering more items to check against our calculations are very close to what it is on live but slightly off.
might be worth waiting till chomp cooks something up.

Seems like Armor plates might have 2 calculations we are doing the first part just not the second part.

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23865) by _**Kaeno**_

**Comment by refringe** [comment]
We found our armor prices are about 10% off live prices, for now we're okay with this as we're unsure of how to simulate BSGs calculation.

Made some changes for keys/armor/jewelry: https://dev.sp-tarkov.com/SPT-AKI/Server/commit/472ab0eabd2dc139fbbf10e39a7988e91df47221

---
[Originally written](https://dev.sp-tarkov.com/SPT/Issues/issues/508#issuecomment-23866) by _**chomp**_
